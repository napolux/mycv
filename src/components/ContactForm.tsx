import React from 'react';
import { Button, Form, Segment, Message, Container, Header } from 'semantic-ui-react'

interface IContactFormState {
  formError: boolean;
  emailSent: boolean
  email: string;
  message: string;
}


export default class ContactForm extends React.Component<{}, Partial<IContactFormState>> {
  constructor(props: {}) {
    super(props)
    this.state = { formError: false, emailSent: false, email: '', message: '' };
    this.handleSubmit = this.handleSubmit.bind(this);
    this.handleChange = this.handleChange.bind(this);
  }

  private handleChange(e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) {
    const obj: { [key: string]: string; } = {};
    obj[e.target.name] = e.target.value;
    this.setState(obj);
  }

  private handleSubmit(e: React.FormEvent<HTMLFormElement>) {
    if (this.state.email !== '' && this.state.message !== '') {
      this.setState({ emailSent: true })
    } else {
      this.setState({ formError: true })
    }
    e.preventDefault();
  }

  render() {

    let formError;

    if (this.state.emailSent === true) {
      return <Container>
        <Message
          success
          header='Email sent!'
          content="I'll get back to you as soon as possible. Thanks."
        />
      </Container>;
    }

    if (this.state.formError === true) {
      formError = <Message
        negative
        header='Error!'
        content="Please compile the form with your email and a message. Thanks."
      />
    }

    return (
      <Container>
        <Segment>
        <Header as='h2'>Hire me!</Header>
          <p>Let's get in touch! Fill the form:</p>
          <Form method="post" onSubmit={this.handleSubmit}>
            <Form.Field>
              <label htmlFor="email">Email:</label>
              <input id="email" type="email" name="email" onChange={this.handleChange} />
            </Form.Field>
            <Form.Field>
              <textarea id="message" name="message" onChange={this.handleChange} />
            </Form.Field>
            {formError}
            <Button type='submit'>Hire me!</Button>
          </Form>
        </Segment>
      </Container>
    );
  }
}