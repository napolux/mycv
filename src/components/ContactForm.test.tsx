import React from 'react';
import ContactForm from './ContactForm';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer
    .create(<ContactForm />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});