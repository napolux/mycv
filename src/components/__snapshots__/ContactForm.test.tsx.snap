// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="ui container"
>
  <div
    className="ui segment"
  >
    <h2
      className="ui header"
    >
      Hire me!
    </h2>
    <p>
      Let's get in touch! Fill the form:
    </p>
    <form
      className="ui form"
      method="post"
      onSubmit={[Function]}
    >
      <div
        className="field"
      >
        <label
          htmlFor="email"
        >
          Email:
        </label>
        <input
          id="email"
          name="email"
          onChange={[Function]}
          type="email"
        />
      </div>
      <div
        className="field"
      >
        <textarea
          id="message"
          name="message"
          onChange={[Function]}
        />
      </div>
      <button
        className="ui button"
        onClick={[Function]}
        type="submit"
      >
        Hire me!
      </button>
    </form>
  </div>
</div>
`;
